### ISSUE TITLE
Can't unpack .NET Reactor 4.5+

### ISSUE THREAD

**Issue by VinaCaptcha** [0xd4d/de4dot#166]
Hello @0xd4d !

Can't unpack .NET Reactor 4.5+, please check my pictures below and any solution for me:
![2018-07-15_012648](https://user-images.githubusercontent.com/33093866/42727327-e558531a-87ce-11e8-838b-f8e4317656eb.jpg)
![2018-07-15_012656](https://user-images.githubusercontent.com/33093866/42727331-eb9705b4-87ce-11e8-84a9-808557b7e844.jpg)
![2018-07-15_012825](https://user-images.githubusercontent.com/33093866/42727332-ef3e8b42-87ce-11e8-9136-36c038467b46.jpg)
Thanks so much!

**Comment by computerline1z** [comment]
Did you have any sample app ?

**Comment by VinaCaptcha** [comment]
@computerline1z here is simple app => https://drive.google.com/file/d/1FRTuDEaIBpJj--j5fdXwflQNYZ8slGh-/view?usp=sharing

**Comment by computerline1z** [comment]
The code flow is simple, seem totally unprotect at all, you could use the default de4dot latest without problem https://transfer.sh/HUvhO/ATRenderPro-cleaned.ex_

**Comment by VinaCaptcha** [comment]
I really using latest de4dot and got this problem! Let me try again. But link not found: https://transfer.sh/HUvhO/ATRenderPro-cleaned.ex

**Comment by computerline1z** [comment]
Did you build the latest project or get the release version ?

**Comment by VinaCaptcha** [comment]
Mình download version mới nhất của de4dot ở đây nè cậu https://ci.appveyor.com/project/0xd4d/de4dot/branch/master/artifa

Có tải project về build thử mà thiếu từa lưa nên build lỗi. Cậu có thể gửi giúp mình bản mới nhất của de4dot với, cái bản mà decode được cái file RenderTools.exe trên ấy, chứ bản mình đang dùng báo lỗi như trên đó. Thanks cậu!

**Comment by computerline1z** [comment]
yes, you could try this https://github.com/computerline1z/somethingusefull/blob/master/de4dot_reactor.7z

**Comment by VinaCaptcha** [comment]
Thanks @computerline1z

**Comment by VinaCaptcha** [comment]
The same error with my version :(
![2018-07-31_223336](https://user-images.githubusercontent.com/33093866/43470045-d4ed0b56-9511-11e8-988f-9aa60229bd1b.jpg)

**Comment by computerline1z** [comment]
Did you try to open the result file ? maybe this because your .net library :(

**Comment by VinaCaptcha** [comment]
Open ATRenderPro-cleaned.exe with error debug.

**Comment by computerline1z** [comment]
Yep, need manual unpack first

**Comment by computerline1z** [comment]
![image](https://user-images.githubusercontent.com/2786420/43474824-1b295b5e-951e-11e8-8a79-2bbb9f3f14a0.png)
The problem cause by Reactor auto resolve type as runtime, you coud see the invalid "call null", just nop out it and everything just fine.

**Comment by VinaCaptcha** [comment]
Ah, có nghĩa là mình sẽ nop dòng code đó trước ở app.exe trước khi decode bằng de4dot hả cậu?

**Comment by computerline1z** [comment]
After decode with de4dot, you open the result file in dnspy, view the file and find the error method, change the mod to IL, and nop out the invalid instruction :)

**Comment by VinaCaptcha** [comment]
Thanks so much :+1:

**Comment by dariuszszyc** [comment]
what if I have dll file instead of exe file ? 
![2018-07-31_2245](https://user-images.githubusercontent.com/41972270/43486210-74984aa2-9513-11e8-85e3-136db67d28ad.png)

**Comment by computerline1z** [comment]
You could try another unpacker with similar protect scheme, could be "de4dot <file> -p crx" or another unpacker :)

**Comment by dariuszszyc** [comment]
I don't understand... if this is packed with .net reactor, why use other like "-p crx" ? Won't it be the same scheme - manual unpack, then de4dot ? 

Unless you mean other reactor unpackers - you know any better than de4dot ?

I can open it with dnspy, but all methods are empty or return true.

**Comment by computerline1z** [comment]
There way to unpack some unknown protector, I write a program use the dll, and load it debug on dnspy, after that, I creat breakpoint at the dll and break in IL code, the common protector unpack the code at runtime, so it must be in the memory, while I trace and see the code has been unpack, I dump the assembly, then use de4dot to cleanup, many protector has same protect scheme, so maybe de4dot can restore it.
